packages/kit/src/runtime/client/client.js

```diff
@@ -1635,6 +1627,7 @@ export function preload_code(pathname) {
 export function push_state(url, state) {
 	if (DEV) {
 		try {
+			// The browser history state's serialization capabilities are roughly equal to devalue.stringify
```

Comment:
the 'roughly' here makes me nervous. why don't we just skip the try-catch? or if we're concerned about the error being overly cryptic, we could wrap the `original_push_state`/`original_replace_state` inside the `try-catch` and then print a friendly error if `error.name === 'DataCloneError'`

Reply:
I tested Map, Set, Date and cyclic dependencies - they were all handled correctly. Custom classes were becoming POJOs. So if anything it can handle more cases than devalue, so I thought it makes sense to keep that dev time check.

Reply:
Ah, so it's less that `devalue.stringify` is a proxy for the browser's own serialization capabilities, and more that we don't _want_ to use the browser's full serialization capabilities because that will cause nasty surprises with certain values. In that case I think we should reflect that more explicitly in the comment:

```suggestion
			// use `devalue.stringify` as a convenient way to ensure we exclude values that can't be properly rehydrated, such as custom class instances
```